### PR TITLE
bugfix/14254-missing-options-arg

### LINKED ISSUE #14254: Highstock plotBands acrossPanes flag has no effect
#### Expected behaviour
Setting `acrossPanes` to false for plotBands in Highstock should only render plot band in single pane.

#### Actual behaviour
Setting `acrossPanes` to false for plotBands in Highstock has no effect. (Possibly related to fix  for #6644 )

#### Live demo with steps to reproduce
https://jsfiddle.net/xkuc5dvt/ (Notice plot band spans all panes)

#### Product version
Highstock 8.2.0

#### Affected browser(s)
Chrome

### PR DESCRIPTION
**Pull request by khlieng**
Fixed #14254, the `plotBands.acrossPanes` option had no effect.

~Fixed #14254, `getPlotBandPath` had no options argument, but its interface did and other callers were using it.~

### CODE REVIEW

File: ts/Core/Axis/PlotLineOrBand.ts [hunk 1]
```diff
@@ -1141,23 +1141,27 @@ extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */ {
...
         from: number,
-        to: number
+        to: number,
+        options?: (Highcharts.AxisPlotBandsOptions|Highcharts.AxisPlotLinesOptions)
     ): SVGPath {
         var toPath = this.getPlotLinePath({
                 value: to,
                 force: true,
-                acrossPanes: (this.options as any).acrossPanes
+                acrossPanes: options?.acrossPanes ?? (this.options as any).acrossPanes
```
Comment (bre1470) [inline ts/Core/Axis/PlotLineOrBand.ts:1159]: Avoid optional markers as they bloat the code. Instead do shorter `&&` `||` checks and remove the `any` cast at the same time.

File: ts/Core/Axis/PlotLineOrBand.ts [hunk 2]
```diff
@@ -1141,23 +1141,27 @@ extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */ {
...
             } as Highcharts.AxisPlotLinePathOptionsObject),
             path = this.getPlotLinePath({
                 value: from,
                 force: true,
-                acrossPanes: (this.options as any).acrossPanes
+                acrossPanes: options?.acrossPanes ?? (this.options as any).acrossPanes
```
Comment (bre1470) [inline ts/Core/Axis/PlotLineOrBand.ts:1164]: Avoid optional markers as they bloat the code. Instead do shorter `&&` `||` checks and remove the `any` cast at the same time.

File: ts/Core/Axis/PlotLineOrBand.ts [hunk 3]
```diff
@@ -1141,23 +1141,27 @@ extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */ {
...
+     *
      * @return {Highcharts.SVGPathArray}
      *         The SVG path definition in array form.
      */
     getPlotBandPath: function (
         this: Highcharts.Axis,
         from: number,
-        to: number
+        to: number,
+        options?: (Highcharts.AxisPlotBandsOptions|Highcharts.AxisPlotLinesOptions)
```
Comment (bre1470) [inline ts/Core/Axis/PlotLineOrBand.ts:1154]: Make use of default values:
```ts
getPlotBandPath: function (
    this: Highcharts.Axis,
    from: number,
    to: number,
    options: (Highcharts.AxisPlotBandsOptions|Highcharts.AxisPlotLinesOptions) = this.options
```

**Review by bre1470 — CHANGES_REQUESTED** [review]
Some optimization is needed.

File: ts/Core/Axis/PlotLineOrBand.ts (hunk 1 above)
Comment (khlieng) [reply ts/Core/Axis/PlotLineOrBand.ts:1159]: How do I get rid of the `any` cast when `AxisOptions` doesn't have `acrossPanes`?

**Review by bre1470 — APPROVED** [review]
Thanks! Looks good. 👍

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks!

Please add a unit test to. We can for example create one plot band which is across panes and one which is not, and verify that the height of the bands is different.

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**